pymc/distributions/timeseries.py

```diff
@@ -142,14 +145,37 @@ class GaussianRandomWalk(distribution.Continuous):
 
     rv_op = gaussianrandomwalk
 
+    def __new__(cls, name, mu=0.0, sigma=1.0, init=None, steps: int = 1, **kwargs):
```

Comment:
Why is overriding `__new__` method needed?

Reply:
`__new__` is the one called when you create a "named" distribution like `pm.Normal("x"...)` instead of calling `.dist()` directoly in `pm.Normal.dist(...)`. This is where it makes sense to check that `init` is not a variable already registered in the model (to avoid double counting)

Reply:
what is happening in `__new__` makes sense, to be more precise why do  we use `__new__` instead of `__init__`?

Reply:
Because these are not real classes, we never initialize pymc distributions only the underlying `rv_op` RandomVariables.